kart/tabular/export.py

```diff
@@ -208,7 +229,7 @@ def table_export(
         if col.data_type == "geometry":
             continue  # Handled separately
         regular_keys.append(col.name)
-        out_layer.CreateField(kart_col_to_ogr_field(col))
+        out_layer.CreateField(kart_schema_col_to_ogr_field_definition(col))
```

Comment:
> [Regarding OGR_DATATYPE_MAP] Don't we have this defined elsewhere for imports?

Yes, sort of.
In practice the forward dict and the reverse dict and/or the forward function and reverse function always end up being slightly different - types which exist in one format but not in the other cannot be perfectly roundtripped and so the mapping is not bijective.

However, I think there is value in having the forward and reverse mapping in one file right next to each other. That file is now `ogr_adapter.py` and I added some basic roundtrip tests in `ogr_adapter_test.py`